### PR TITLE
CT-3107 clone other feature specs working

### PR DESCRIPTION
**Pull request by Polsonby**
## Description
This PR adds the following user stories to the Complaints correspondence type. 

- Feature: Searching for cases
- Feature: Progressing a case
- Feature: Adding data requests
- Feature: Closing a case
- Feature: Generating acknowledgement letters
- Feature: Logging data received
- Feature: Previewing a cover page
- Feature: Adding note to case history

This should be pretty much all of the clone functionality working - last things to do (not in this PR) are reports and any miscellaneous specs that have slipped through!

## Self-review checklist

* [ ] (1) Quick stakeholder demo done OR
* [ ] (2) ...bug with before and after screenshots
* [x] (3) Tests passing
* [x] (4) Branch ready to be merged (not work in progress)
* [x] (5) No superfluous changes in diff
* [x] (6) No TODO's without new ticket numbers
* [x] (7) PR Prefixed with ticket number e.g. `CT-7654 ...`

### Screenshots

image

### Related JIRA tickets
The main story is here: https://dsdmoj.atlassian.net/browse/CT-3107
The subtasks for these are:
https://dsdmoj.atlassian.net/browse/CT-3147
https://dsdmoj.atlassian.net/browse/CT-3152
https://dsdmoj.atlassian.net/browse/CT-3145
https://dsdmoj.atlassian.net/browse/CT-3146
https://dsdmoj.atlassian.net/browse/CT-3148
https://dsdmoj.atlassian.net/browse/CT-3149
https://dsdmoj.atlassian.net/browse/CT-3150
https://dsdmoj.atlassian.net/browse/CT-3153

### Manual testing instructions
Given an offender SAR Complaint case, you should be able to edit everything, progress the case, add notes, add and edit data requests, update the page totals, generate cover page, generate acknowledgement and dispatch letters.

### CODE REVIEW

File: spec/features/cases/offender_sar_complaint/adding_note_to_case_history_spec.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+require 'rails_helper'
+
+feature 'When viewing an offender sar case' do
```
Comment (willmcb) [inline spec/features/cases/offender_sar_complaint/adding_note_to_case_history_spec.rb:3]: Small thing but may be confusing in future - we should probably change "offender sar case" to "offender sar complaint case" here. Possibly in the from "case" to "complaint" in the file name as well?

File: spec/features/cases/offender_sar_complaint/adding_note_to_case_history_spec.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  given(:responder)         { find_or_create :branston_user }
+  given(:offender_sar_complaint) { create(:offender_sar_complaint).decorate }
+
+  scenario 'adding a note' do
+    login_as responder
+    cases_show_page.load(id: offender_sar_complaint.id)
+
+    expect(cases_show_page).to have_content "Add a note to this case"
```
Comment (willmcb) [inline spec/features/cases/offender_sar_complaint/adding_note_to_case_history_spec.rb:11]: Again here with case -> complaint

File: spec/features/cases/offender_sar_complaint/case_progressing_spec.rb [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    login_as manager
+    cases_page.load
+    CaseClosure::MetadataSeeder.seed!
+  end
+
+  after do
+    CaseClosure::MetadataSeeder.unseed!
+  end
+
+  scenario 'progressing an offender sar case', js: true do
```
Comment (willmcb) [inline spec/features/cases/offender_sar_complaint/case_progressing_spec.rb:19]: Case -> complaint or complaint case :)

File: spec/features/cases/offender_sar_complaint/case_progressing_spec.rb [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    expect(cases_close_page).to be_displayed
+    cases_close_page.fill_in_date_responded(offender_sar_complaint.received_date)
+    click_on "Continue"
+
+    expect(cases_closure_outcomes_page).not_to be_displayed
+
+    expect(cases_show_page).to be_displayed
+    expect(cases_show_page).to have_content "Closed"
+    expect(cases_show_page).to have_content "Send dispatch letter"
+    # TODO - pending decision on closure outcomes https://dsdmoj.atlassian.net/browse/CT-2502
```
Comment (willmcb) [inline spec/features/cases/offender_sar_complaint/case_progressing_spec.rb:61]: I guess is old and relates to offender SARs? Probably best we remove this comment 👍

**Review by willmcb — APPROVED** [review]
👍 this looks great - few small non-blocking comments, nice job :)

**Comment by willmcb** [comment]
User flow looks good tested everything I could think of 👍

File: spec/features/cases/offender_sar_complaint/case_progressing_spec.rb (hunk 3 above)
Comment (Polsonby) [reply spec/features/cases/offender_sar_complaint/case_progressing_spec.rb:19]: I think for our internal terminology in the app it's fine to call complaint cases "cases", it's what they literally are. So I wouldn't change nouns around except where it's necessary for disambiguation. In the UI, it's a fair question too and even there I'd argue for sticking with "case" personally

File: app/views/cases/offender_sar_complaint/_date_responded_step.html.slim [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+  span.visually-hidden
+    = t('common.case.header_case_number')
+  = " #{@case.number}"
+
+= render partial: 'layouts/header'
+
+div class="case-#{@correspondence_type_key}"
+  / NOTE: this uses the step mechanism to change the date_responded after a case has closed.
+  / this does not appear in the list of steps for new cast creation as found in
+  / OffenderSARCaseForm -> def steps method
```
[the remark this answers is not in the document]
Comment (Polsonby) [reply app/views/cases/offender_sar_complaint/_date_responded_step.html.slim:14]: Yes, we are - just! Thanks, updated :)

**Pull request merged**